core/include/gnuradio-4.0/Port.hpp

```
@@ -226,7 +226,9 @@ Follows the ISO 80000-1:2022 Quantities and Units conventions:
             if (std::holds_alternative<T>(value)) {
                 member = std::get<T>(value);
             } else {
-                throw gr::exception("invalid-argument: incorrect type for " + key);
+                if constexpr (!isNoexcept) {
```

Should perhaps either drop the exception in favour of `std::expected` (preferred for internal GR4-lib API) or at least provide another path and not just swallow the error. 

Also, please use `std::source_location` in the method parameter and `gr::exception`/`std::expected<T, Error>` to aid localising errors.

---

I’ve made `PortMetaInfo::update` always `noexcept` and it now returns `std::expected<void, Error>`.
The open question was how the constructor should handle failures. 

Possible strategies were: 1) throwing constructor, 2) static factory returning `expected`, 3) record error, 4) ignore, 5) terminate.

I implemented  a throwing constructor to match `Block` constructor behavior when parameters are invalid.
```cpp
    explicit PortMetaInfo(std::initializer_list<std::pair<const std::string, pmtv::pmt>> initMetaInfo) noexcept(false) //
        : PortMetaInfo(property_map{initMetaInfo.begin(), initMetaInfo.end()}) {}
    explicit PortMetaInfo(const property_map& metaInfo) noexcept(false) {
        if (auto res = update(metaInfo); !res) {
            throw gr::exception(res.error().message, res.error().sourceLocation);
        }
    }
```